core/trino-main/src/main/java/io/trino/operator/project/PageProcessorMetrics.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.project;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.plugin.base.metrics.DurationTiming;
+import io.trino.spi.metrics.Metric;
+import io.trino.spi.metrics.Metrics;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class PageProcessorMetrics
+{
+    private static final String FILTER_TIME = "Filter CPU time";
+    private static final String PROJECTION_TIME = "Projection CPU time";
+
+    private long filterTimeNanos;
+    private boolean hasFilter;
+    private long projectionTimeNanos;
+    private boolean hasProjection;
+
+    public void recordFilterTime(long startNanos)
```

Comment:
The new shape is better, but I have a slight concern about the fact that both methods now look the same singnature-wise but behave very differently. I think this could be improved by renaming this method to `recordFilterTimeSince` to indicate that it's taking care of the "until" measurement.

Once we fix how we're measuring timing for filters (per my other comment), we can change it to look like the `recordProjectionTime` method below.